pubsub.go

```diff
@@ -203,6 +222,9 @@ func NewPubSub(ctx context.Context, h host.Host, rt PubSubRouter, opts ...Option
 	h.Network().Notify((*PubSubNotif)(ps))
 
 	ps.val.Start(ps)
+	if err := ps.disc.Start(ps); err != nil {
+		return nil, err
+	}
```

Comment:
@vyzo I just changed the `Start` function to return an error since creating a cache can throw an error. In practice this should never happen though.

Should I move things around here (e.g. move `ps.disc.Start` at least before `val.Start` if not earlier so that any other goroutines also get cancelled?

Reply:
I moved it up.